apps/personal/contacts/main.py

```diff
@@ -4,57 +4,113 @@
 
 import os
 
-from address_book import AddressBook, ZPUI_HOME, Contact
+from address_book import AddressBook, ZPUI_HOME
+from contact import Contact
 from apps import ZeroApp
 from helpers import setup_logger
-from ui import NumberedMenu, Listbox
-from vcard_converter import VCardContactConverter
+from ui import NumberedMenu, Listbox, Menu, LoadingIndicator, Printer, DialogBox
+from distutils.spawn import find_executable
 
 logger = setup_logger(__name__, "info")
 
+# FIXME: load from config module?
+VDIRSYNCER_CONFIG = '~/.config/vdirsyncer/config'
+VDIRSYNCER_VCF_DIRECOTRY = '~/.local/share/vdirsyncer/contacts/contacts'
 
 class ContactApp(ZeroApp):
     def __init__(self, i, o):
         super(ContactApp, self).__init__(i, o)
         self.menu_name = "Contacts"
         self.address_book = AddressBook()
         self.menu = None
+        self.vdirsyncer_executable = find_executable('vdirsyncer')
 
     def on_start(self):
+        self.reload()
+
+    def reload(self):
         self.address_book.load_from_file()
-        self.menu = NumberedMenu(self.create_menu_content(), self.i, self.o, prepend_numbers=False)
+        self.menu = NumberedMenu(self.build_main_menu_content(), i=self.i,
+                                 o=self.o, prepend_numbers=False)
         self.menu.activate()
 
-    def create_menu_content(self):
+    def build_main_menu_content(self):
         all_contacts = self.address_book.contacts
-        return [[c.short_name(), lambda x=c: self.create_contact_page(x)] for c in all_contacts]
+        menu_entries = []
+        menu_entries.append(["|| Actions", lambda: self.open_actions_menu()])
+        for c in all_contacts:
+            menu_entries.append([c.short_name(), lambda x=c:
+                                 self.open_contact_details_page(x)])
+
+        return menu_entries
 
-    def create_contact_page(self, contact):
+    def open_contact_details_page(self, contact):
         # type: (Contact) -> None
-        contact_attrs = [getattr(contact, a) for a in contact.get_filled_attributes()]
+        contact_attrs = [getattr(contact, a) for a in
+                         contact.get_filled_attributes()]
         Listbox(i=self.i, o=self.o, contents=contact_attrs).activate()
```

Comment:
okay, I see now - this was in the original code. If this worked before, I have no idea how this did work =D Sorry for that, I will take care to review others' code better in the future.

Reply:
Fixed by bb19598.